### PR TITLE
Add Multiplatform Logging

### PR DESCRIPTION
**Pull request by marcellogalhardo**
While contributing to the multiplatform client code, I missed a logging solution. I'm not sure if `Napier` is the best solution out there but it is the only one I found supporting all platforms we support.

https://github.com/AAkira/Napier

What do you folks think of it?

### CODE REVIEW

**Comment by joreilly** [comment]
No strong opinion....I'm using Kermit on some other projects but Napier also popular as well.

**Comment by martinbonnin** [comment]
Apollo doesn't do logging but exposes APIs instead (see also https://publicobject.com/2022/05/01/eventlisteners-are-good/). If I need to debug something quickly, I usually do `println()` and (hopefully!) remove it before committing.

But Napier works too 👍

**Review by martinbonnin — APPROVED** [review]

File: androidApp/src/main/java/dev/johnoreilly/confetti/ConfettiApplication.kt [hunk 1]
```diff
@@ -32,7 +30,10 @@ class ConfettiApplication : Application(), ImageLoaderFactory {
         } else {
             Firebase.crashlytics.setCrashlyticsCollectionEnabled(false)
         }
-        
+
+        // Initialize Logging.
+        Napier.base(DebugAntilog())
```
Comment (joreilly) [inline androidApp/src/main/java/dev/johnoreilly/confetti/ConfettiApplication.kt:35]: Does something equivalent have to be done for Wear and iOS clients/

Comment (marcellogalhardo) [reply androidApp/src/main/java/dev/johnoreilly/confetti/ConfettiApplication.kt:35]: For iOS:
```
fun debugBuild() {
    Napier.base(DebugAntilog())
}
```

We would also need to do it on WearOS + Compose Desktop.

**Comment by marcellogalhardo** [comment]
@martinbonnin I'm familiar with the article, great one btw.

Does `print` in a multiplatform code (i.e., SessionsViewModel) print on Logcat console? It wasn't showing for me (but I don't have much experience with KMP anyway, so I may have been doing it wrong?). 🤔

File: androidApp/src/main/java/dev/johnoreilly/confetti/ConfettiApplication.kt (hunk 1 above)
Comment (marcellogalhardo) [reply androidApp/src/main/java/dev/johnoreilly/confetti/ConfettiApplication.kt:35]: I will include it on Wear and Desktop, I don't have iOS tooling to compile and test, tho.

File: compose-desktop/src/main/kotlin/main.kt [hunk 2]
```diff
@@ -36,6 +54,11 @@ private val dateService = JvmDateService()
 fun main() = application {
     val windowState = rememberWindowState()
 
+    LaunchedEffect(key1 = this) {
```
Comment (marcellogalhardo) [inline compose-desktop/src/main/kotlin/main.kt:57]: @joreilly I have no clue if that is the "correct way" to do it on Compose Desktop. Suggestions?

File: androidApp/src/main/java/dev/johnoreilly/confetti/ConfettiApplication.kt (hunk 1 above)
Comment (marcellogalhardo) [reply androidApp/src/main/java/dev/johnoreilly/confetti/ConfettiApplication.kt:35]: Included on Android, Wear, Desktop. I currently don't have the tooling to compile and test iOS, someone will need to do it there. 😄

**Comment by martinbonnin** [comment]
> Does print in a multiplatform code (i.e., SessionsViewModel) print on Logcat console?

It should although it bypasses the logcat tags (and maybe other things?) so it will end up with a `System.out` tag. If you're filtering on `System.out` you should see it (which is also a good way to filter out the noise :) )

File: compose-desktop/src/main/kotlin/main.kt (hunk 2 above)
Comment (joreilly) [reply compose-desktop/src/main/kotlin/main.kt:57]: I'm definitely no expert on Compose for Desktop but that seems ok to me at least

**Pull request merged**